includes/woopay/class-woopay-adapted-extensions.php

```diff
@@ -171,9 +171,38 @@ public function get_extension_data() {
 			];
 		}
 
+		if ( $this->is_affiliate_for_woocommerce_enabled() ) {
+			$extension_data[ 'affiliate-for-woocommerce' ] = [
+				'affiliate-user' => afwc_get_referrer_id(),
+			];
+		}
+
 		return $extension_data;
 	}
 
+	/**
+	 * Update order extension data after finishing
+	 * an order on WooPay, this usually is needed
+	 * for extensions which uses cookies when an
+	 * order is finished.
+	 *
+	 * @param int $order_id The successful WooPay order.
+	 */
+	public function update_order_extension_data( $order_id ) {
+		if ( ! empty( $_GET['affiliate'] ) && // phpcs:ignore WordPress.Security.NonceVerification
+			$this->is_affiliate_for_woocommerce_enabled()
+		) {
+			$affiliate_id = wc_clean( wp_unslash( $_GET['affiliate'] ) ); // phpcs:ignore WordPress.Security.NonceVerification
```

Comment:
The affiliate ID is always based on the user ID, right?  Should we coerce this to an integer?

Reply:
Fixed on https://github.com/Automattic/woocommerce-payments/pull/7291/commits/d976182614707218207b70a60918db298ff0651f.